### ISSUE TITLE
Update of conda from 4.7.10 to 4.7.12 fails due to setuptools being marked for removal

### ISSUE THREAD

**Issue by monocongo** [conda/conda#9431]
## Current Behavior

I am prompted to update conda, but when I try to do so using the suggested update command I get an error indicating that setuptools is marked for deletion but cannot be removed since it's a dependency for conda. 

### Steps to Reproduce

```
$ conda install pandas
Collecting package metadata (current_repodata.json): done
Solving environment: - 
done


==> WARNING: A newer version of conda exists. <==
  current version: 4.7.10
  latest version: 4.7.12

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.

$ conda update -n base -c defaults conda
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/james/miniconda3

  added / updated specs:
    - conda


The following packages will be REMOVED:

  bzip2-1.0.8-h7b6447c_0
  libarchive-3.3.3-h5d8350f_5
  libxml2-2.9.9-hea5a465_1
  lz4-c-1.8.1.2-h14c3975_0
  lzo-2.10-h49e0be7_2
  pip-19.1.1-py37_0
  python-libarchive-c-2.8-py37_11
  setuptools-41.0.1-py37_0
  wheel-0.33.4-py37_0
  zstd-1.3.7-h0b5b093_0

The following packages will be UPDATED:

  asn1crypto                                  0.24.0-py37_0 --> 1.2.0-py37_0
  ca-certificates                               2019.5.15-0 --> 2019.10.16-0
  certifi                                  2019.6.16-py37_0 --> 2019.9.11-py37_0
  cffi                                1.12.3-py37h2e261b9_0 --> 1.13.1-py37h2e261b9_0
  chardet                                      3.0.4-py37_1 --> 3.0.4-py37_1003
  conda                                       4.7.10-py37_0 --> 4.7.12-py37_0
  conda-package-han~                          1.3.11-py37_0 --> 1.6.0-py37h7b6447c_0
  cryptography                           2.7-py37h1ba5d50_0 --> 2.8-py37h1ba5d50_0
  openssl                                 1.1.1c-h7b6447c_1 --> 1.1.1d-h7b6447c_3
  pysocks                                      1.7.0-py37_0 --> 1.7.1-py37_0
  sqlite                                  3.29.0-h7b6447c_0 --> 3.30.1-h7b6447c_0
  tqdm                                          4.32.1-py_0 --> 4.36.1-py_0


Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: failed

RemoveError: 'setuptools' is a dependency of conda and cannot be removed from
conda's operating environment.
```

## Expected Behavior

conda should be updated to the latest version (4.7.12)

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
$ conda info

     active environment : None
            shell level : 0
       user config file : /home/james/.condarc
 populated config files : /home/james/.condarc
          conda version : 4.7.10
    conda-build version : not installed
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.1
       base environment : /home/james/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/james/miniconda3/pkgs
                          /home/james/.conda/pkgs
       envs directories : /home/james/miniconda3/envs
                          /home/james/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.10 requests/2.22.0 CPython/3.7.3 Linux/4.15.0-65-generic linuxmint/19.2 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
$ conda config --show-sources
==> /home/james/.condarc <==
auto_activate_base: False
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
$ conda list --show-channel-urls
# packages in environment at /home/james/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
asn1crypto                0.24.0                   py37_0    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
ca-certificates           2019.5.15                     0    defaults
certifi                   2019.6.16                py37_0    defaults
cffi                      1.12.3           py37h2e261b9_0    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.7.10                   py37_0    defaults
conda-package-handling    1.3.11                   py37_0    defaults
cryptography              2.7              py37h1ba5d50_0    defaults
idna                      2.8                      py37_0    defaults
libarchive                3.3.3                h5d8350f_5    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
libxml2                   2.9.9                hea5a465_1    defaults
lz4-c                     1.8.1.2              h14c3975_0    defaults
lzo                       2.10                 h49e0be7_2    defaults
ncurses                   6.1                  he6710b0_1    defaults
openssl                   1.1.1c               h7b6447c_1    defaults
pip                       19.1.1                   py37_0    defaults
pycosat                   0.6.3            py37h14c3975_0    defaults
pycparser                 2.19                     py37_0    defaults
pyopenssl                 19.0.0                   py37_0    defaults
pysocks                   1.7.0                    py37_0    defaults
python                    3.7.3                h0371630_0    defaults
python-libarchive-c       2.8                     py37_11    defaults
readline                  7.0                  h7b6447c_5    defaults
requests                  2.22.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37h14c3975_0    defaults
setuptools                41.0.1                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.29.0               h7b6447c_0    defaults
tk                        8.6.8                hbc83047_0    defaults
tqdm                      4.32.1                     py_0    defaults
urllib3                   1.24.2                   py37_0    defaults
wheel                     0.33.4                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zlib                      1.2.11               h7b6447c_3    defaults
zstd                      1.3.7                h0b5b093_0    defaults
```
</p></details>

**Comment by jennirinker** [comment]
I've had this issue for months now, conda 4.7.5.

**Comment by monocongo** [comment]
I seem to have solved this by using `conda-forge` rather than the default channel.

**Comment by msarahan** [comment]
This is fixed in conda 4.7.11, but old versions will have trouble updating.  `conda update setuptools` helps it because it makes setuptools an explicit dependency.  Sorry for the inconvenience, but we can't retroactively fix conda 4.7.5.  Your best bet is to use Miniconda, https://repo.anaconda.com/miniconda/, which has 4.7.12 in its latest version.
